Adding more options to the SSL configuration (#2039)

LannyRipple opened the pull request:
* Provides ClientSSLConfig FromJavaxNetSsl to allow configuring client SSLContext using a javax.net.ssl.KeyManagerFactory and/or a javax.net.ssl.TrustManagerFactory.
* Only partially addresses #2039.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zio/zio-http?pullRequest=2583) <br/>All committers have signed the CLA.

LannyRipple commented:
Attempt to address some of Client SSL Configuration using javax.net.ssl.  Leaving this as draft for guidance and criticism.

Things I don't like or am not sure of:
* My naming of variables and types.
* ClientSSLConfig.FromJavaxNetSsl has odd builder() entry points.  Needed to ensure a valid configuration.  Any better way?
* Implemented tests but ClientHttpsSpec (and my extension) ignores all tests.

LannyRipple committed:
Adding more options to the SSL configuration (#2039)

* Provides ClientSSLConfig FromJavaxNetSsl to allow configuring
  client SSLContext using a javax.net.ssl.KeyManagerFactory and/or
  a javax.net.ssl.TrustManagerFactory.
* Only partially addresses #2039.

987Nabil commented:
@LannyRipple I changed the secrets from `String` to `Secret` so you have to solve some conflicts please 🙂

987Nabil commented:
And the `ClientHttpsSpec` is broken, we had before a config for a website that is no longer available via https. Therefore we deactivated the tests to fix CI. You can remove the `@@ ignore` to test it. If you can fix the test too, we would be very grateful 🙏

LannyRipple commented:
@987Nabil Thanks for eyeballing and the feedback.  Will fix these up.

LannyRipple committed:
rebase to main

jdegoes commented:
@LannyRipple Looks good to merge! Please re-open when conflicts are fixed and the build is passing. Thank you!

LannyRipple commented:
Thanks.  Will do when I can find the time for tests (even if just local).
